crates/iota-data-ingestion-core/src/lib.rs

```diff
@@ -60,7 +61,7 @@ pub trait Worker: Send + Sync {
         Some(sequence_number)
     }
 
-    fn preprocess_hook(&self, _: CheckpointData) -> Result<()> {
+    fn preprocess_hook(&self, _: CheckpointData) -> anyhow::Result<()> {
```

Comment:
Here too, since we opted in making a library-specific error, I reckon the trait should reflect it as well.

Reply:
[fixed](https://github.com/iotaledger/iota/pull/5152/commits/0152e28a813a195a15d5e9a230a0b6a01a415109#diff-fe85c8baebf1df63db20121a40711b2e88a9bfe28cf2cd71041b22dba5802afaR20-R29)